javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -1003,8 +1003,8 @@ public Character get(int index) {
     }
 
     @Override
-    public int indexOf(Character element, int from) {
-        return back.indexOf(element, from);
+    public int indexOf(Object element, int from) {
```

Comment:
I think `element` can stay of type `Character` because `Character` is a final class. The we do not need the cast in the line below.